### PR TITLE
181 delete user from backend

### PR DESCRIPTION
**Pull request by hussojo**
Added backend logic to deleting user and couple postman test.

### CODE REVIEW

File: backend/User/views.py [hunk 1]
```diff
@@ -79,7 +80,17 @@ def delete(self, request, user_id):
 			return Response(status=status.HTTP_404_NOT_FOUND)
```
Comment (dorianjagusch) [inline backend/User/views.py:80]: I know it's not part of this PR, but shouldn't we first check that the request is well-formed and only if it is log the user out/ send a 404?

File: backend/User/views.py [hunk 2]
```diff
@@ -79,7 +80,17 @@ def delete(self, request, user_id):
...
 		except:
 			return Response(status=status.HTTP_400_BAD_REQUEST)
-		user.delete()
+		user.username = get_random_string(length=30)
```
Comment (dorianjagusch) [inline backend/User/views.py:83]: I don't know what was discussed yesterday but we might want do to something else than a random string here like "deleted user" or something, I would say (unless there's a good reason to randomise it)

**Review by dorianjagusch — APPROVED** [review]

**Comment by dorianjagusch** [comment]
Once my PR comes through you can "easily" add a confirmation popup to our page with authentication. Both modals are basically complete for that. We might need to discuss about the services and how to connect them in a flow, cause that can be slightly counter intuitive (decoupling wohooo)

**Review by dorianjagusch — APPROVED** [review]

File: backend/User/views.py [hunk 3]
```diff
@@ -79,7 +80,17 @@ def delete(self, request, user_id):
...
+		user.set_password(get_random_string(length=30))
+		user.is_active = False
+		user.is_staff = False
+		user.is_superuser = False
+		user.insertTS = None
+		user.last_login = None
+		user.is_online = False
+		# profile_picture = None
```
Comment (merituulie) [inline backend/User/views.py:91]: the profile_picture comment can be taken out, itll be in its own table!

**Review by merituulie — APPROVED** [review]

File: backend/User/views.py (hunk 1 above)
Comment (hussojo) [reply backend/User/views.py:80]: As far as I understand the methods used are already making sure user is logged in and also the owner. So the 404 should never even happen in this case 🤷‍♀️

**Review by dorianjagusch — APPROVED** [review]

**Pull request merged**